File: NicoJK/TVTComment/IPC/RawIPCMessage.cpp
#include "stdafx.h"
#include "RawIPCMessage.h"

namespace TVTComment
{
	std::string RawIPCMessage::ToString() const
	{
		std::string ret;
		for (const std::string &content : this->Contents)
			ret += content+" ";
		ret.pop_back();
		return this->MessageName + ret;
	}
}

File: TVTComment/Model/IPC/RawIPCMessage.cs
using System.Collections.Generic;

namespace TVTComment.Model.IPC
{
    /// <summary>
    /// 生のIPCMessage
    /// </summary>
    /// <remarks>
    /// <see cref="MessageName"/>と<see cref="Contents"/>はrecord separatorとunit separatorを含んではならない
    /// </remarks>
    class RawIPCMessage
    {
        public string MessageName { get; set; }
        public IEnumerable<string> Contents { get; set; }

        public override string ToString()
        {
            return $"MessageName={MessageName},Contents={{{string.Join(" ", Contents)}}}";
        }
    }
}
